### ISSUE TITLE
Improve LightGCN model recall from 10.6% to 20%+

### ISSUE THREAD

**Issue by Pranav1011** [Pranav1011/Recommendation-System#11]
## Overview

Current LightGCN model achieves **Recall@10 = 10.59%**, which is below state-of-the-art for MovieLens 25M. Industry benchmarks show 15-25% is achievable. This issue tracks experiments to improve model performance.

## Current Performance Baseline

```
LightGCN (Best Model):
- NDCG@10:    5.27%
- Recall@10:  10.59%  ⬅️ Target for improvement
- MAP@10:     3.52%
- Hit Rate:   32.22%
- Coverage:   6.76%

Training: 137,883 users, 34,461 movies, 20M ratings
```

## Target Performance

- **Primary Goal**: Recall@10 ≥ 20% (+89% improvement)
- **Secondary Goal**: NDCG@10 ≥ 8% (+52% improvement)
- **Constraint**: Maintain latency < 100ms P95

### 1. Hyperparameter Tuning
- [ ] Increase embedding dim from 64 → 128/256
- [ ] Tune layer depth (current: 3 layers, try 4-5)
- [ ] Experiment with learning rate schedule
- [ ] Try different batch sizes (current: 1024)

### 2. Graph Construction Improvements  
- [ ] Lower rating threshold from 4.0 → 3.5 (more edges)
- [ ] Add implicit feedback (view counts, time spent)
- [ ] Experiment with edge weighting strategies

### 3. Advanced Techniques
- [ ] Multi-task learning (rating prediction + ranking)
- [ ] Contrastive learning objectives
- [ ] Graph augmentation (dropout, masking)
- [ ] Negative sampling improvements

### 4. Ensemble Approaches
- [ ] Weighted ensemble with Two-Tower (current: failed at 0.85%)
- [ ] Time-based ensemble (recent vs historical patterns)
- [ ] User segment-specific models

### 5. Cold Start Handling
- [ ] Content-based features for new items
- [ ] User demographic features
- [ ] Metadata embeddings (genres, tags, etc.)

## Experiment Tracking

Use MLflow to log all experiments with:
- Model config
- Training metrics
- Evaluation results
- Inference latency

## Acceptance Criteria

- ✅ Recall@10 ≥ 20%
- ✅ NDCG@10 ≥ 8%
- ✅ P95 latency < 100ms
- ✅ Coverage ≥ 5%
- ✅ All experiments logged in MLflow

## Resources

- **Papers**: 
  - LightGCN: https://arxiv.org/abs/2002.02126
  - NGCF: https://arxiv.org/abs/1905.08108
- **Benchmarks**: 
  - RecBole leaderboard
  - MovieLens 25M papers

## Estimated Effort

2-4 weeks of experimentation

## Priority

**HIGH** - Model performance is core value proposition
